src/ApiConnectors/ApiOptions.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace PhpTwinfield\ApiConnectors;
+
+/**
+ * Class ApiOptions
+ * @package PhpTwinfield\ApiConnectors
+ */
+final class ApiOptions
+{
+    /**
+     * @var string[] exception messages that should be retried
+     */
+    private $retriableExceptionMessages = [
+        "SSL: Connection reset by peer",
+        "Your logon credentials are not valid anymore. Try to log on again."
+    ];
+    private $maxRetries = 3;
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(?array $messages = null, ?int $maxRetries = null)
+    {
+        if ($messages !== null) {
+            $this->validateMessages($messages);
+            $this->retriableExceptionMessages = $messages;
+        }
+        if ($maxRetries !== null) {
+            $this->validateMaxRetries($maxRetries);
+            $this->maxRetries = $maxRetries;
+        }
+    }
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    private function validateMaxRetries(int $maxRetries): void
+    {
+        if ($maxRetries < 0) {
+            throw new \InvalidArgumentException('The max retries should be a positive integer.');
+        }
+    }
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    private function validateMessages(array $messages): void
+    {
+        $exceptions = [];
```

Comment:
I don't think there's a lot of value in overcomplicating this with multiple exception strings. I'd keep the code simple but robust:

```
foreach ($messages as $message) {
    if (trim($message) === '') {
        throw ...
    }
}
```